File: SolutionsPython/003.py
# Find the largest prime factor of 600851475143
def isPrime(num):
    for x in range(2, int(num**0.5)):
        if not num % x:
            return False

    return True


def isFactor(fac, n):
    if not n % fac:
        return True

    return False


def largest_prime_factor(n):
    i = int(n**0.5)

    while not (isPrime(i, n) and isFactor(i, n)):
        i -= 1

    return i


print(largest_prime_factor(600851475143.)) 
# 6857
# Runtime: 1.919s

File: SolutionsR/003.R
# What is the largest prime factor of the number 600851475143?
number = 600851475143

isPrimeFac <- function(num) {
  if (number %% num == 0) {
    for (x in 2:round(num**0.5)) {
      if (num %% x == 0) {
        return(FALSE)
      }
    }
    return(TRUE)
  }
  return(FALSE)
}

i <- round(number**0.5)
while (isPrimeFac(i) == FALSE){
  i <- i - 1
}

print(i)
# 6857